packages/nestjs-authorization/src/permission.provider.ts

```diff
@@ -1,5 +1,9 @@
-import { Permission } from './permission';
+import { DefaultPermissionSet } from 'permission-set';
 
-export abstract class PermissionProvider {
-  abstract findPermissions(user: any): Promise<Permission[]>;
+export interface PermissionProvider {
+  getPermissionSetForUser(user: any): Promise<DefaultPermissionSet>;
```

Comment:
Nothing here says yet that it must be user who we're requesting permissions for. Do we want to stick with that notion

Reply:
how would an alternative look like?